src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -229,7 +229,12 @@ string EncodeUrl (Uri url)
 
 			return DoProcessRequest (request, javaUrl, httpConnection, cancellationToken, redirectState);
 		}
-		
+
+		async Task DisconnectAsync (HttpURLConnection httpConnection)
+		{
+			await Task.Run (() => httpConnection?.Disconnect ()).ConfigureAwait (continueOnCapturedContext: false);
```

Comment:
This should just be `return Task.Run (() => httpConnection?.Disconnect ());` and `async` should be removed from the method signature.